### ISSUE TITLE
Exclusion of rule misses out comma spacing correction in PHPCBF

### ISSUE THREAD

**Issue by u01jmg3** [squizlabs/PHP_CodeSniffer#1151]
Using the following rule/exclusion and PHP, PHPCBF misses out performing its comma spacing correction.

``` xml
<rule ref="Squiz.Arrays.ArrayDeclaration">
    <exclude name="Squiz.Arrays.ArrayDeclaration.SingleLineNotAllowed" />
</rule>
```

``` php
$x = array(1,2);
```
## 

Suggestion is to move the [comma spacing correction code](https://github.com/squizlabs/PHP_CodeSniffer/blob/e11f25393c3e5795ff87f2ccb2c156c63d6a13b3/CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php#L257-L297) above the [PHPCBF fixing logic](https://github.com/squizlabs/PHP_CodeSniffer/blob/e11f25393c3e5795ff87f2ccb2c156c63d6a13b3/CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php#L240) in `ArrayDeclarationSniff` to allow the spacing to be corrected even if the `SingleLineNotAllowed` sniff is muted.

**Comment by u01jmg3** [comment]
Sorry to be pushy but will this enhancement be implemented at some point soon? Thanks

**Comment by gsherwood** [comment]
> Sorry to be pushy but will this enhancement be implemented at some point soon?

Probably not. I barely have enough time to get bugs fixed and getting a 3.0 release is my top priority.

**Comment by u01jmg3** [comment]
No worries - if you are ever looking for extra hands I would be happy to help

**Comment by gsherwood** [comment]
Closing in favour of #582 , which would break out the array sniff and allow for this, and other, customisations.
